### PR TITLE
Add support for squash_commit_sha in Merge Requests

### PR DESCRIPTION
**Pull request by marcojahn**

### CODE REVIEW

**Comment by gmessner** [comment]
@marcojahn Thank you for your contribution.

I have just one small request, I just audited the JSON being returned and validated in the documentation that there is another field missing ```references```.  This filed is an object defined by this JSON:
```
    "references": {
      "short": "!1",
      "relative": "!1",
      "full": "my-group/my-project!1"
    }
```
Could you add the ```References``` class to ```org.gitlab4j.api.models``` and add it as a field in the ```MergeRequest``` class?

**Comment by marcojahn** [comment]
Added, please have a review.

File: src/main/java/org/gitlab4j/api/models/References.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    private String relative;
+    private String full;
+
+    public String get_short() {
+        return _short;
+    }
+
+    public void set_short(String _short) {
+        this._short = _short;
+    }
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/models/References.java]: @marcojahn 
Change these to ```getShort()``` and ```setShort(String short)```.

Comment (marcojahn) [reply src/main/java/org/gitlab4j/api/models/References.java]: @gmessner done pls review

**Comment by gmessner** [comment]
@marcojahn GitLab4J-API 4.14.17 is now available with this enhancement.

**Pull request merged**